model-optimizer/extensions/ops/tensor_iterator.py

```diff
@@ -347,6 +347,51 @@ def ti_type_infer(node):
 
         ti_graph.remove_nodes_from([node.id for node in fake_input_const_nodes])
 
+    @staticmethod
+    def find_iterations_count_for_output(ti_node, output_rec):
+        def check_field(record, field):
+            return field in record and record[field] is not None
+        iterations_count = 1
+        # 1. check if we need concatenate iteration results for given output
+        if not check_field(output_rec, 'axis'):
+            # in this case we dont concat outputs, so iterations count is not needed really
+            return iterations_count
+
+        # 2. check if given output record contains values for 'end', so iterations count can be calculated from this record
+        if check_field(output_rec, 'end') and output_rec['end'] != -1 and \
+                check_field(output_rec, 'start') and output_rec['start'] != -1:
+            # get iterations count from output record
+            iterations_count = (output_rec['end'] - output_rec['start']) / output_rec['stride']
+            return iterations_count
```

Comment:
why do we rely on only output_record here during iteration number computation? We can compute it if we walk though all output records in output map so the concrete output record is not needed here

Reply:
Output can be cut in any way, so check of other outputs is not useful. We check only current output if it contains concrete limits for output cutting/gathering